File: benchmarks/python/objects.py
class Point:
    __slots__ = ("x", "y")

    def __init__(self, x, y):
        self.x = x
        self.y = y

    def sum(self):
        return self.x + self.y


points = [Point(i, i * 2) for i in range(1_000_000)]
print(sum(p.sum() for p in points))

File: benchmarks/rust/objects.rs
struct Point { x: i64, y: i64 }
impl Point { fn sum(&self) -> i64 { self.x + self.y } }
fn main() {
    let mut points: Vec<Box<Point>> = Vec::new();
    for i in 0..1_000_000i64 { points.push(Box::new(Point { x: i, y: i * 2 })); }
    let total: i64 = points.iter().map(|p| p.sum()).sum();
    println!("{}", total);
}
